syncr_backend/tracker_util.py

```diff
@@ -0,0 +1,33 @@
+import socket
+
+import bencode  # type: ignore
+
+from syncr_backend.constants import TRACKER_BUFFER_SIZE
+from syncr_backend.constants import TRACKER_ERROR_RESULT
+
+
+def send_request_to_tracker(request: list, ip: str, port: int) -> dict:
+    """
+    Creates a connection with the tracker and sends a given request to the
+    tracker and returns the response
+    :param port: port where tracker is serving
+    :param ip: ip of tracker
+    :param request: ['POST'/'GET', node_id, public key if 'POST']
+    :return: tracker response or an artificial response in case of timeout
+    """
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        s.connect((ip, port))
+        s.send(bencode.encode(request))
+        data = s.recv(TRACKER_BUFFER_SIZE)
```

Comment:
```
data = b''
while True:
    temp = s.recv(size)
    if len(temp) == 0:
        break
    data += temp
```

Reply:
I tried this and it led to stalling out, I will look into how to do this properly over the weekend so that i can get my commit in